core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -332,6 +334,8 @@
         NullPolicy.STRICT);
     defineMethod(CONCAT, BuiltInMethod.STRING_CONCAT.method,
         NullPolicy.STRICT);
+    defineMethod(CONCAT2, BuiltInMethod.STRING_CONCAT.method, NullPolicy.STRICT);
```

Comment:
Maybe there is no need to register `CONCAT2` into `RexImpTable`. In runtime, Calcite can automatically find the concat function. You can have a try.

Reply:
Still keep defineMethod for CONCAT2 for looking up method clearly.